src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -307,10 +307,15 @@ class PatternMatcher extends MiniPhaseTransform with DenotTransformer {thisTrans
       // TODO Deal with guards?
 
       def isSwitchableType(tpe: Type): Boolean = {
-        (tpe isRef defn.IntClass) ||
-        (tpe isRef defn.ByteClass) ||
-        (tpe isRef defn.ShortClass) ||
-        (tpe isRef defn.CharClass)
+        val actualTpe = tpe match {
+          case t @ AnnotatedType(inner, _) => inner
```

Comment:
Agreed, I ran the tests and looked at the call sites and we should be fine with putting it in `isRef`